sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -55,10 +55,11 @@ private[sql] class SharedState(
 
   SharedState.setFsUrlStreamHandlerFactory(sparkContext.conf, sparkContext.hadoopConfiguration)
 
-  private val (conf, hadoopConf) = {
+  private[sql] val (conf, hadoopConf) = {
     // Load hive-site.xml into hadoopConf and determine the warehouse path which will be set into
     // both spark conf and hadoop conf avoiding be affected by any SparkSession level options
-    SharedState.loadHiveConfFile(sparkContext.conf, sparkContext.hadoopConfiguration)
+    SharedState.loadHiveConfFile(
+      sparkContext.conf, sparkContext.hadoopConfiguration, initialConfigs)
```

Comment:
how is it related to RESET?

Reply:
This is kind of a hidden bug here, If we create a SparkSession like this: 
```scala
SparkSession.builder.sparkContext(sc).config("spark.sql.warehouse.dir", "abc").getOrCreate()
```
The `"spark.sql.warehouse.dir", "abc"` is used by the SessionCatalog correctly, but we do not set it in the cloned conf in SharedState. It causes the default database may use a wrong path here https://github.com/apache/spark/blob/f253fad00c14376e950804849481fa6252cd8154/sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala#L135
And If we want to use it for RESET, we also need this config to be kept here.

Reply:
I updated the test case a bit, FYI https://github.com/yaooqinn/sugar/blob/master/src/main/scala/com/netease/mammut/spark/training/sql/WarehouseSCBeforeSS.scala#L67

'data2' is expected